src/uct/tcp/sockcm/sockcm_iface.c

```diff
@@ -94,29 +124,165 @@ static uct_iface_ops_t uct_sockcm_iface_ops = {
 {
-    ucs_trace("sockcm_iface_client_start_next_ep not implemented");
+    ucs_status_t status;
+    uct_sockcm_ep_t *ep, *tmp;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+
+    /* try to start an ep from the pending eps list */
+    ucs_list_for_each_safe(ep, tmp, &iface->pending_eps_list, list_elem) {
+        status = uct_sockcm_ep_set_sock_id(iface, ep);
+        if (status != UCS_OK) {
+            continue;
+        }
+
+        ucs_list_del(&ep->list_elem);
+        ep->is_on_pending = 0;
+
+        uct_sockcm_ep_set_failed(&iface->super.super, &ep->super.super, status);
+    }
+
+    UCS_ASYNC_UNBLOCK(iface->super.worker->async);
+}
+
+static void uct_sockcm_iface_event_handler(int fd, void *arg)
+{
+    ucs_debug("not implemented yet");
 }
 
 static UCS_CLASS_INIT_FUNC(uct_sockcm_iface_t, uct_md_h md, uct_worker_h worker,
                            const uct_iface_params_t *params,
                            const uct_iface_config_t *tl_config)
 {
+    uct_sockcm_iface_config_t *config = ucs_derived_of(tl_config, uct_sockcm_iface_config_t);
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    ucs_status_t status;
+    int ret = 0;
+    struct sockaddr *param_sockaddr;
+    int param_sockaddr_len;
+
+    UCT_CHECK_PARAM(params->field_mask & UCT_IFACE_PARAM_FIELD_OPEN_MODE,
+                    "UCT_IFACE_PARAM_FIELD_OPEN_MODE is not defined");
+
+    UCT_CHECK_PARAM((params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER) ||
+                    (params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_CLIENT),
+                    "Invalid open mode %zu", params->open_mode);
+
+    UCT_CHECK_PARAM(!(params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER) ||
+                    (params->field_mask & UCT_IFACE_PARAM_FIELD_SOCKADDR),
+                    "UCT_IFACE_PARAM_FIELD_SOCKADDR is not defined for UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER");
+
     UCS_CLASS_CALL_SUPER_INIT(uct_base_iface_t, &uct_sockcm_iface_ops, md, worker,
                               params, tl_config
                               UCS_STATS_ARG((params->field_mask &
                                              UCT_IFACE_PARAM_FIELD_STATS_ROOT) ?
                                             params->stats_root : NULL)
                               UCS_STATS_ARG(UCT_SOCKCM_TL_NAME));
 
+    if (self->super.worker->async == NULL) {
+        ucs_error("sockcm must have async != NULL");
+        return UCS_ERR_INVALID_PARAM;
+    }
+    if (self->super.worker->async->mode == UCS_ASYNC_MODE_SIGNAL) {
+        ucs_warn("sockcm does not support SIGIO");
+    }
+
+    self->listen_fd = -1;
+
+    if (params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER) {
+
+        param_sockaddr = (struct sockaddr *) params->mode.sockaddr.listen_sockaddr.addr;
+        param_sockaddr_len = params->mode.sockaddr.listen_sockaddr.addrlen;
+
+        status = ucs_socket_create(param_sockaddr->sa_family, SOCK_STREAM,
```

Comment:
the code looks very common with `uct_tcp_iface_listener_init`, maybe define a common function for them?